### ISSUE TITLE
Error with Layout/MultilineMethodCallIndentation

### ISSUE THREAD

**Issue by ccutrer** [rubocop/rubocop#12265]
## Expected behavior

No errors

## Actual behavior

An error:

```
Scanning /Users/cody/src/canvas-lms/test.rb
An error occurred while Layout/MultilineMethodCallIndentation cop was inspecting /Users/cody/src/canvas-lms/test.rb:5:4.
undefined method `join' for nil:NilClass

          node.loc.dot.join(node.loc.selector)
                      ^^^^^
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/layout/multiline_method_call_indentation.rb:187:in `semantic_alignment_base'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/layout/multiline_method_call_indentation.rb:155:in `alignment_base'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/layout/multiline_method_call_indentation.rb:91:in `offending_range'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/mixin/multiline_expression_indentation.rb:20:in `on_send'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/commissioner.rb:107:in `public_send'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/commissioner.rb:107:in `block (2 levels) in trigger_responding_cops'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/commissioner.rb:171:in `with_cop_error_handling'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/commissioner.rb:106:in `block in trigger_responding_cops'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/commissioner.rb:105:in `each'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/commissioner.rb:105:in `trigger_responding_cops'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/commissioner.rb:69:in `on_send'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:154:in `on_block'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:154:in `on_def'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/commissioner.rb:71:in `on_def'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:154:in `on_class'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/commissioner.rb:71:in `on_class'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:20:in `walk'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/commissioner.rb:87:in `investigate'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/team.rb:156:in `investigate_partial'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cop/team.rb:98:in `investigate'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:345:in `block in inspect_file'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:344:in `each'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:344:in `flat_map'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:344:in `inspect_file'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:287:in `block in do_inspection_loop'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:321:in `block in iterate_until_no_changes'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:314:in `loop'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:314:in `iterate_until_no_changes'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:283:in `do_inspection_loop'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:164:in `block in file_offenses'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:189:in `file_offense_cache'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:163:in `file_offenses'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:154:in `process_file'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:135:in `block in each_inspected_file'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:134:in `each'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:134:in `reduce'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:134:in `each_inspected_file'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:120:in `inspect_files'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/runner.rb:73:in `run'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cli/command.rb:11:in `run'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cli/environment.rb:18:in `run'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cli.rb:118:in `run_command'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cli.rb:125:in `execute_runners'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cli.rb:51:in `block in run'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cli.rb:77:in `profile_if_needed'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/lib/rubocop/cli.rb:43:in `run'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/exe/rubocop:19:in `block in <top (required)>'
/Users/cody/.rubies/ruby-3.1.3/lib/ruby/3.1.0/benchmark.rb:311:in `realtime'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.57.0/exe/rubocop:19:in `<top (required)>'
/Users/cody/src/canvas-lms/bin/rubocop:16:in `load'
/Users/cody/src/canvas-lms/bin/rubocop:16:in `<main>'
```

The error occurs twice for the same file.

## Steps to reproduce the problem

test.rb, with default rubocop config:

```ruby
# frozen_string_literal: true

class Test
  def validate_urls
    (
      [url] + Lti::ResourcePlacement::PLACEMENTS.map do |p|
        settings[p]&.with_indifferent_access&.fetch("url", nil) ||
        settings[p]&.with_indifferent_access&.fetch("target_link_uri", nil)
      end
    )
      .compact
      .map { |u| validate_url(u) }
  end
  private :validate_urls
end
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
cody@P6C2THJ7GJ src % rubocop -V
1.57.0 (using Parser 3.2.2.4, rubocop-ast 1.29.0, running on ruby 3.1.3) [arm64-darwin21]
```
